### PR TITLE
Add mutating policies danger explanations

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Fix https://github.com/kubewarden/docs/issues/335.

## Test

See https://deploy-preview-336--silly-bunny-8cedd0.netlify.app/next/tasksDir/mutating-policies.

### CODE REVIEW

File: docs/tasksDir/mutating-policies.md [hunk 1]
```diff
@@ -11,22 +11,61 @@ Mutating policies receive an object request and rebuild this incoming object
...
+### Misconfigured mutating policies together with 3rd party Kubernetes Controllers can get stuck in an infinite loop
 
-However, if you set the `ClusterAdmissionPolicy.mutating` field to `false`,
-the mutated requests will be rejected.
-For example, create the following `ClusterAdmissionPolicy` with the `mutating` field set to `true`:
+:::danger
+Mutating policies return requests that proceed through the Kubernetes API. If
+there are other Kubernetes Controllers that listen for those same resources,
+they may mutate them back in a follow-up request. This could lead to an
+infinite feedback loop of mutations.
```
Comment (viccuad) [inline docs/tasksDir/mutating-policies.md:42]: Maybe it's worth to add a specific example of a bad `spec.rules` in this case?

**Review by jhkrug — APPROVED** [review]
lgtm

File: docs/tasksDir/mutating-policies.md [hunk 2]
```diff
@@ -11,22 +11,61 @@ Mutating policies receive an object request and rebuild this incoming object
...
+1. Have the policy inspect low level resources (e.g: Pod). Different kind of
+   Kubernetes resources (be them native or CRDs) can create Pods. By having the
+   policy target Pod objects, there's the guarantee all the Pods are going to be
+   compliant.
+   However, this could lead to confusion among end users of the cluster: their
+   high level Kubernetes resources would be successfully created, but they would
+   stay in a non reconciled state. For example, a Deployment creating a
+   non-compliant Pod would be created, but it would never have all its replicas
+   running. The end user would have to do some debugging to finally understand why
+   this is happening.
```
Comment (flavio) [inline docs/tasksDir/mutating-policies.md:59]: This should be dropped. That applies to validating policies, not to mutating ones.

File: docs/tasksDir/mutating-policies.md [hunk 3]
```diff
@@ -11,22 +11,61 @@ Mutating policies receive an object request and rebuild this incoming object
...
+2. Have the policy inspect higher order resource (e.g. Deployment). The end
+   users get immediate feedback about the rejections. However, there's still
+   the chance that some non compliant pods are created by another high level
+   resource (be it native to Kubernetes, or a CRD).
```
Comment (flavio) [inline docs/tasksDir/mutating-policies.md:63]: This applies to validating policies, not to mutating ones.

We could say that targeting the higher level resource could however lead to loops if a controller is managing them. That includes for example controllers of GitOps solutions (like fleet, flux, argo, ...) or other 3rd party controllers that translate their own CRDs into Deployment objects.

**Review by flavio — CHANGES_REQUESTED** [review]

**Comment by viccuad** [comment]
@flavio tackled the suggestions, please re-review.

**Review by jhkrug — APPROVED** [review]
lgtm

**Review by jvanz — APPROVED** [review]

**Pull request merged**